docs/backend-system/building-backends/08-migrating.md

```diff
@@ -1063,3 +1063,289 @@ backend.start();
+
+##### Postgres
+
+An installation of the Search plugin using the Postgres search engine will look as follows:
+
+```ts title="packages/backend/src/index.ts"
+const backend = createBackend();
+
+// Other plugins...
+
+/* highlight-add-start */
+backend.add(import('@backstage/plugin-search-backend/alpha'));
+backend.add(import('@backstage/plugin-search-backend-module-pg/alpha'));
+/* highlight-add-end */
+```
+
+##### Elasticsearch
+
+A basic installation of the Search plugin using the Elasticsearch search engine will look as follows:
+
+```ts title="packages/backend/src/index.ts"
+const backend = createBackend();
+
+// Other plugins...
+
+/* highlight-add-start */
+backend.add(import('@backstage/plugin-search-backend/alpha'));
+backend.add(
+  import('@backstage/plugin-search-backend-module-elasticsearch/alpha'),
+);
+/* highlight-add-end */
+```
+
+#### Search Collators
+
+The following sections outline how you add the Catalog and TechDocs search collators
+
+##### Catalog
+
+A basic installation of the Search plugin with the Catalog collator will look as follows:
+
+```ts title="packages/backend/src/index.ts"
+const backend = createBackend();
+
+// Other plugins...
+
+/* highlight-add-start */
+backend.add(import('@backstage/plugin-search-backend/alpha'));
+backend.add(import('@backstage/plugin-search-backend-module-catalog/alpha'));
+/* highlight-add-end */
+```
+
+##### TechDocs
+
+A basic installation of the Search plugin with the TechDocs collator will look as follows:
+
+```ts title="packages/backend/src/index.ts"
+const backend = createBackend();
+
+// Other plugins...
+
+/* highlight-add-start */
+backend.add(import('@backstage/plugin-search-backend/alpha'));
+backend.add(import('@backstage/plugin-search-backend-module-techdocs/alpha'));
+/* highlight-add-end */
+```
+
+### The Permissions Plugin
+
+A basic installation of the Permission plugin will look as follows:
+
+```ts title="packages/backend/src/index.ts"
+const backend = createBackend();
+
+// Other plugins...
+
+/* highlight-add-start */
+backend.add(import('@backstage/plugin-permission-backend/alpha'));
+backend.add(
+  import('@backstage/plugin-permission-backend-module-allow-all-policy'),
+);
+/* highlight-add-end */
+```
+
+> Note: the above includes a default allow all policy, if that is not what you want do not add the 2nd line.
+
+#### Custom Permission Policy
+
+In order to add your own permission policy you'll need to do the following:
+
+```ts
+import { createBackendModule } from '@backstage/backend-plugin-api';
+import { BackstageIdentityResponse } from '@backstage/plugin-auth-node';
+import {
+  PolicyDecision,
+  AuthorizeResult,
+} from '@backstage/plugin-permission-common';
+import {
+  PermissionPolicy,
+  PolicyQuery,
+} from '@backstage/plugin-permission-node';
+import { policyExtensionPoint } from '@backstage/plugin-permission-node/alpha';
+
+class ExampleAllowAllPermissionPolicy implements PermissionPolicy {
```

Comment:
It's probably unfortunate to explicitly name this `AllowAll`. Imagine that this will be copy pasted a lot and then only the contents of the `handle` method updated. Maybe name the class `CustomPermissionPolicy` to begin with?

Reply:
Yeah, that makes sense as I do `policy.setPolicy(new CustomPermissionPolicy());` so this doesn't even match what I have across the example.

Reply:
Fixed up 👍